schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/AvroSnapshotSerializer.java

```diff
@@ -50,10 +51,24 @@ public AvroSnapshotSerializer(ISchemaRegistryClient schemaRegistryClient) {
         super(schemaRegistryClient);
     }
 
+    @Override
+    public void init(Map<String, ?> config) {
+        super.init(config);
+
+        protocolVersion = (Byte) ((Map<String, Object>) config).getOrDefault(SERDES_PROTOCOL_VERSION, SchemaVersionProtocolHandlerRegistry.CURRENT_PROTOCOL);
+
+        if(SchemaVersionProtocolHandlerRegistry.get().getSerDesProtocolHandler(protocolVersion) == null) {
+            throw new IllegalArgumentException("SchemaVersionProtocolHandler with protocol version " + protocolVersion + " does not exist");
+        }
+    }
+
     protected byte[] doSerialize(Object input, SchemaIdVersion schemaIdVersion) throws SerDesException {
         try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
-            writeProtocolId(CURRENT_PROTOCOL_VERSION, byteArrayOutputStream);
-            writeSchemaVersion(schemaIdVersion, byteArrayOutputStream);
+            SchemaVersionProtocolHandlerRegistry
+                    .get()
+                    .getSerDesProtocolHandler(protocolVersion)
+                    .handleSchemaVersionSerialization(byteArrayOutputStream, schemaIdVersion);
+
             writeContentPayload(input, byteArrayOutputStream);
```

Comment:
I wanted to start with how versioned schema identifier are encoded as part of protocol definition. Added javadoc comment explaining this in `SchemaVersionProtocolHandler`. I am fine in adding the actual serialization of payload as part of protocol in the next step.